src/ripple/basics/impl/CheckLibraryVersions.cpp

```diff
@@ -81,18 +78,32 @@ void checkVersion(std::string name, std::string required, std::string actual)
 
 void checkBoost(std::string version)
```

Comment:
Even if these weren't single-use functions, I think that passing by value makes sense here. Rationale: we call this function (and other, similar ones in this file) in one of two ways: 

1. With a `char const*` argument: in that case there's no fundamental difference between having the argument be `std::string const&` instead of `std::string`; either way, a `std::string` must be constructed.
2. With an actual `std::string` as the argument: in that case, move semantics ~will~ should generate optimal code.